### PR TITLE
To move ILogger in constructor for all triggers

### PR DESCRIPTION
**Pull request by surgupta-msft**
To move ILogger in constructor for all triggers and remove static from class and Run function

### CODE REVIEW

File: Functions.Templates/Templates/BlobTrigger-CSharp-Isolated/BlobTriggerCSharp.cs [hunk 1]
```diff
@@ -5,14 +5,19 @@
 
 namespace Company.Function
 {
-    public static class BlobTriggerCSharp
+    public class BlobTriggerCSharp
     {
+        private readonly ILogger<BlobTriggerCSharp> log;
```
Comment (soninaren) [inline Functions.Templates/Templates/BlobTrigger-CSharp-Isolated/BlobTriggerCSharp.cs:10]: rename to `_logger`

File: Functions.Templates/Templates/BlobTrigger-CSharp-Isolated/BlobTriggerCSharp.cs [hunk 2]
```diff
@@ -5,14 +5,19 @@
...
+
+        public BlobTriggerCSharp(FunctionContext context)
+        {
+            this.log = context.GetLogger("BlobTriggerCSharp");
```
Comment (soninaren) [inline Functions.Templates/Templates/BlobTrigger-CSharp-Isolated/BlobTriggerCSharp.cs:14]: i don't think `this.` is needed

File: Functions.Templates/Templates/BlobTrigger-CSharp-Isolated/BlobTriggerCSharp.cs [hunk 3]
```diff
@@ -5,14 +5,19 @@
...
+        }
+
         [Function("BlobTriggerCSharp")]
-        public static void Run([BlobTrigger("PathValue/{name}", Connection = "ConnectionValue")] string myBlob, string name,
-            FunctionContext context)
+        public void Run([BlobTrigger("PathValue/{name}", Connection = "ConnectionValue")] string myBlob, string name)
         {
-            var logger = context.GetLogger("BlobTriggerCSharp");
-            logger.LogInformation($"C# Blob trigger function Processed blob\n Name: {name} \n Data: {myBlob}");
+            log.LogInformation($"C# Blob trigger function Processed blob\n Name: {name} \n Data: {myBlob}");
```
Comment (soninaren) [inline Functions.Templates/Templates/BlobTrigger-CSharp-Isolated/BlobTriggerCSharp.cs:20]: `_logger.LogInformation`

File: Functions.Templates/Templates/BlobTrigger-CSharp/BlobTriggerCSharp.csx [hunk 4]
```diff
@@ -1,4 +1,4 @@
-public static void Run(Stream myBlob, string name, ILogger log)
+public void Run(Stream myBlob, string name)
 {
     log.LogInformation($"C# Blob trigger function Processed blob\n Name:{name} \n Size: {myBlob.Length} Bytes");
```
Comment (soninaren) [inline Functions.Templates/Templates/BlobTrigger-CSharp/BlobTriggerCSharp.csx:3]: This would cause compile error. Where is log coming from.

**Review by soninaren — CHANGES_REQUESTED** [review]
Have requested changes on one template. same would go for other templates.

File: Functions.Templates/Templates/BlobTrigger-CSharp/BlobTriggerCSharp.csx (hunk 4 above)
Comment (surgupta-msft) [reply Functions.Templates/Templates/BlobTrigger-CSharp/BlobTriggerCSharp.csx]: I will discuss about the changes needed in .csx files from you and Fabio offline and then update the code here.

**Review by soninaren — APPROVED** [review]

**Pull request merged**